### ISSUE TITLE
Tests broken ("internal error: No guest options available for arch 'x86_64'",)

### ISSUE THREAD

**Issue by cd1** [kimchi-project/kimchi#597]
Commit 640b257 has broken the tests on Fedora 21. With that commit, the tests end quicker than usual (probably because not all of them are executed) with an error message and without the usual tests report.

I'm running Kimchi commit 2f6a5d2 on Fedora 21.

## How to reproduce ##

1. Download Kimchi by cloning the Git repository or by extracting a source tarball.
2. Install its dependencies.
3. Go to the working directory and type the following commands:
  - `./autogen.sh --system`
  -  `make`
  - `sudo make check`

## Actual behaviour ##

After step (3), the following output is displayed on the screen:

```
[...I omitted the initial output as I don't think it's relevant here...]

...WebSocket server settings:
  - Listen on :64667
  - Flash security policy server
  - Web server. Web root: /home/vianac/kimchi/ui/pages/websockify
  - SSL/TLS support
  - Deny non-SSL/TLS connections
  - proxying from :64667 to targets in /var/lib/kimchi/vnc-tokens

[17/Feb/2015:12:14:17] ENGINE Bus STARTING
[17/Feb/2015:12:14:17] ENGINE Started monitor thread 'Autoreloader'.
[17/Feb/2015:12:14:17] ENGINE Serving on 127.0.0.1:45562
*** Running feature tests ***
127.0.0.1 - - [17/Feb/2015:12:14:17] "HEAD /images/icon-fedora.png HTTP/1.1" 200 4449 "" ""
127.0.0.1 - - [17/Feb/2015:12:14:17] "GET /images/icon-fedora.png HTTP/1.1" 206 4449 "" ""
127.0.0.1 - - [17/Feb/2015:12:14:17] "HEAD /images/icon-fedora.png HTTP/1.1" 200 4449 "" ""
127.0.0.1 - - [17/Feb/2015:12:14:17] "GET /images/icon-fedora.png HTTP/1.1" 206 4449 "" ""
libvirt: Domain Config error : internal error: No guest options available for arch 'x86_64'
[17/Feb/2015:12:14:17] ENGINE Error in 'start' listener <bound method CapabilitiesModel._set_capabilities of <kimchi.model.config.CapabilitiesModel object at 0x7f9482eeef50>>
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 197, in publish
    output.append(listener(*args, **kwargs))
  File "/home/vianac/kimchi/src/kimchi/model/config.py", line 71, in _set_capabilities
    self.metadata_support = FeatureTests.has_metadata_support(conn)
  File "/home/vianac/kimchi/src/kimchi/model/featuretests.py", line 200, in has_metadata_support
    'arch': arch})
  File "/home/vianac/kimchi/src/kimchi/model/libvirtconnection.py", line 66, in wrapper
    ret = f(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/libvirt.py", line 3484, in defineXML
    if ret is None:raise libvirtError('virDomainDefineXML() failed', conn=self)
libvirtError: internal error: No guest options available for arch 'x86_64'

[17/Feb/2015:12:14:17] ENGINE Shutting down due to error in start listener:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 235, in start
    self.publish('start')
  File "/usr/lib/python2.7/site-packages/cherrypy/process/wspbus.py", line 215, in publish
    raise exc
ChannelFailures: libvirtError("internal error: No guest options available for arch 'x86_64'",)

[17/Feb/2015:12:14:17] ENGINE Bus STOPPING
[17/Feb/2015:12:14:17] ENGINE HTTP Server cherrypy._cpwsgi_server.CPWSGIServer(('127.0.0.1', 45562)) shut down
[17/Feb/2015:12:14:17] ENGINE Stopped thread 'Autoreloader'.
[17/Feb/2015:12:14:17] ENGINE Bus STOPPED
[17/Feb/2015:12:14:17] ENGINE Bus EXITING
[17/Feb/2015:12:14:17] ENGINE Bus EXITED
Makefile:479: recipe for target 'check-local' failed
make[3]: *** [check-local] Error 70
make[3]: Leaving directory '/home/vianac/kimchi/tests'
Makefile:353: recipe for target 'check-am' failed
make[2]: *** [check-am] Error 2
make[2]: Leaving directory '/home/vianac/kimchi/tests'
Makefile:355: recipe for target 'check' failed
make[1]: *** [check] Error 2
make[1]: Leaving directory '/home/vianac/kimchi/tests'
Makefile:415: recipe for target 'check-recursive' failed
make: *** [check-recursive] Error 1
```

## Expected behaviour ##

After step (3), all the tests should execute successfully and a test report should be displayed in the end.

**Comment by alinefm** [comment]
Duplicated of #591

**Comment by cd1** [comment]
I am still hitting this issue using the latest Git commit (1a43119), even with the fix for #591 applied.

**Comment by cd1** [comment]
Running the following commands will cause the issue:

```
$ cd tests
$ sudo PYTHONPATH=../src:../ python -m unittest  -v test_model.ModelTests.test_template_create test_model_network.NetworkTests.test_get_networks
```

I tested on Fedora 21 and Ubuntu 14.10, the issue is reproducible on both systems.

**Comment by alinefm** [comment]
I am looking into it.
The problem happens because we are mixing tests with the real model and those using the libvirt Test Driver in test_model.py

**Comment by alinefm** [comment]
Patch sent to ML: [Kimchi-devel] [PATCH] Fix issue #597: Get the right arch when	using libvirt Test driver
